src/Features/LanguageServer/Protocol/Handler/Configuration/DidChangeConfigurationNotificationHandler.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CommonLanguageServerProtocol.Framework;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Configuration
+{
+    internal partial class DidChangeConfigurationNotificationHandler : ILspServiceNotificationHandler<LSP.DidChangeConfigurationParams>, IOnInitialized
+    {
+        private readonly ILspLogger _lspLogger;
+        private readonly IGlobalOptionService _globalOptionService;
+        private readonly IClientLanguageServerManager _clientLanguageServerManager;
+        private readonly Guid _registrationId;
+
+        public DidChangeConfigurationNotificationHandler(ILspLogger logger, IGlobalOptionService globalOptionService, IClientLanguageServerManager clientLanguageServerManager)
+        {
+            _lspLogger = logger;
+            _globalOptionService = globalOptionService;
+            _clientLanguageServerManager = clientLanguageServerManager;
+            _registrationId = Guid.NewGuid();
+        }
+
+        public bool MutatesSolutionState => false;
+
+        public bool RequiresLSPSolution => true;
+
+        [LanguageServerEndpoint(Methods.WorkspaceDidChangeConfigurationName)]
+        public Task HandleNotificationAsync(DidChangeConfigurationParams request, RequestContext requestContext, CancellationToken cancellationToken)
```

Comment:
Since this only implements 1 request, put the method on the type like this - https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/Diagnostics/DocumentPullDiagnosticHandler.cs,18